src/device-registry/utils/generate-filter.js

```diff
@@ -981,7 +981,7 @@ const generateFilter = {
         filter["_id"] = ObjectId(progress_id);
       }
       if (user_id) {
-        filter["user_id"] = ObjectId(user_id);
+        filter["user_id"] = user_id;
```

Comment:
Hi @BenjaminSsempala , isn't this the `user_id `being added to the filter object?

Reply:
@Baalmart Yes but this filter is for KYA progresses, while the other one is for KYA lessons

Reply:
@Baalmart Yes but this filter is for KYA progresses, while the other one is for KYA lessons

Reply:
Cool! Thanks!